packages/teams-js/src/public/dialog.ts

```diff
@@ -210,6 +217,121 @@ export namespace dialog {
     }
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert AdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
+   *
+   * @internal
+   */
+  export function getDialogInfoFromAdaptiveCardDialogInfo(adaptiveCardDialogInfo: AdaptiveCardDialogInfo): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: adaptiveCardDialogInfo.card,
+      height: adaptiveCardDialogInfo.size ? adaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: adaptiveCardDialogInfo.size ? adaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: adaptiveCardDialogInfo.title,
+    };
+    return dialogInfo;
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert BotAdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
+   *
+   * @internal
+   */
+  export function getDialogInfoFromBotAdaptiveCardDialogInfo(
+    botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+  ): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: botAdaptiveCardDialogInfo.card,
+      height: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
+  }
+  /**
+   * Subcapability for interacting with adaptive card dialogs
+   */
+  export namespace adaptiveCard {
+    /**
+     * Allows app to open an adaptive card based dialog.
+     *
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param adaptiveCardDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
+     */
+    export function open(adaptiveCardDialogInfo: AdaptiveCardDialogInfo, submitHandler?: DialogSubmitHandler): void {
+      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      const dialogInfo: DialogInfo = getDialogInfoFromAdaptiveCardDialogInfo(adaptiveCardDialogInfo);
+      sendMessageToParent('tasks.startTask', [dialogInfo], (err: string, result: string | object) => {
+        submitHandler?.({ err, result });
+        removeHandler('messageForParent');
```

Comment:
We are not registering any handler here either. Should we delete this removeHandler?